vpr/src/util/vpr_utils.cpp

```diff
@@ -501,6 +501,45 @@ std::tuple<ClusterNetId, int, int> find_pb_route_clb_input_net_pin(ClusterBlockI
     return std::tuple<ClusterNetId, int, int>(clb_net_idx, curr_pb_pin_id, clb_net_pin_idx);
 }
 
+std::vector<ClusterPinId> cluster_pins_connected_to_atom_pin(AtomPinId atom_pin) {
```

Comment:
I am not sure what you are using this for, or how large the cluster pins vector will be, but returning a vector will create a vector copy much like a pass by value. You can have the user pass an empty vector for you to fill if this is a critical function. There may be ways around this copy, but I am not familiar with them or they may not have nice style.